### PR TITLE
fix: Fixed `stat.paddle.mean` for all backends

### LINKED ISSUE #28082: Fix Frontend Failing Test: jax - stat.paddle.mean
To-do List: https://github.com/unifyai/ivy/issues/27496

### LINKED ISSUE #28083: Fix Frontend Failing Test: numpy - stat.paddle.mean
To-do List: https://github.com/unifyai/ivy/issues/27497

### LINKED ISSUE #28084: Fix Frontend Failing Test: tensorflow - stat.paddle.mean
To-do List: https://github.com/unifyai/ivy/issues/27499

### LINKED ISSUE #28085: Fix Frontend Failing Test: torch - stat.paddle.mean
To-do List: https://github.com/unifyai/ivy/issues/27498

### LINKED ISSUE #28086: Fix Frontend Failing Test: paddle - stat.paddle.mean
To-do List: https://github.com/unifyai/ivy/issues/27500

### PR DESCRIPTION
**Pull request by Sai-Suraj-27**
# PR Description
Fixed `stat.paddle.mean` for all backends

image

## Related Issue
Closes #28082
Closes #28083
Closes #28084
Closes #28085
Closes #28086

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [x] Did you run your tests and are your tests passing?
- [x] Did pre-commit not fail on any check?
- [x] Did you follow the steps we provided?

### CODE REVIEW

**Comment by Sarvesh-Kesharwani** [comment]
image

**Comment by Sai-Suraj-27** [comment]
> image image please work on fixing these test failures

Hey, @Sarvesh-Kesharwani thanks for the review. I'm not sure, how many examples you set while testing, I tested with 25, 50, 100 and 150 examples.

image
All, the backends are passing with 50 examples.

Although when I tested with 100 examples, only `torch` backend is failing and when I tested with 150 examples only `numpy` backend is failing.

image

But, it is actually not any problem with code, it's just the accuracy of the result. You can see that the test is failing due to a minute difference of 10^(-3) difference in the outputs from the 2 backends, Due to this small difference some tests are failing when we set the number of examples too high.
 
image

So, I guess we can merge this, no need to change anything. Please let me know your opinion.
Thank you.

**Review by KareemMAX — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_stat.py
Comment (KareemMAX) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_stat.py:29]: Oh, I thought it won't unpack the rest like JS (sorry have been off Python for a while). But why don't we use this?
```suggestion
    input_dtype, x, axis = dtype_and_x[:2]
```
Not sure if I'm slicing correctly though

**Comment by KareemMAX** [comment]
Also, regarding the current issues with the failing tests. They shouldn't fail even if the code appears right, the precision issue should be handled accordingly @Sai-Suraj-27. And make sure examples cited by @Sarvesh-Kesharwani are passing

File: ivy_tests/test_ivy/test_frontends/test_paddle/test_stat.py
Comment (Sai-Suraj-27) [reply ivy_tests/test_ivy/test_frontends/test_paddle/test_stat.py:29]: Done👍🏻

**Comment by Sai-Suraj-27** [comment]
> Also, regarding the current issues with the failing tests. They shouldn't fail even if the code appears right, the precision issue should be handled accordingly @Sai-Suraj-27. And make sure examples cited by @Sarvesh-Kesharwani are passing

Hey, @KareemMAX I know precision should be handled but since the test was passing in CI also. I thought it's okay. Actually this is a clear example of this [2nd case](https://unify.ai/docs/ivy/overview/contributing/error_handling.html#error-handling) in the docs.
I have updated the test, and tested it for 150 examples, and it is successfully passing for all the backends.
image

Thanks @Sarvesh-Kesharwani for spotting this 😄.

**Review by KareemMAX — APPROVED** [review]
LGTM, waiting for @Sarvesh-Kesharwani's approval to merge

**Comment by Sai-Suraj-27** [comment]
Hey, @Sarvesh-Kesharwani. Can you please look into the PR once and merge it if you think it's good 😄.
Thank you.

**Comment by Sarvesh-Kesharwani** [comment]
Okay let me test once, then I'll merge it.

**Comment by Sarvesh-Kesharwani** [comment]
Hi @Sai-Suraj-27,
LGTM now,
Thanks for the hard work.

**Pull request merged**
